braindecode/visualization/confusion_matrices.py

```diff
@@ -0,0 +1,239 @@
+            if x != y or (not with_f1_score):
+                ax.annotate(
+                    "\n\n{:4.1f}%".format(
+                        normed_conf_mat[x][y] * 100),
+                    xy=(y, x),
+                    horizontalalignment='center',
+                    verticalalignment='center', fontsize=10,
+                    color=textcolor,
+                    fontweight=this_font_weight)
+            else:
+                assert x == y
+                precision = confusion_mat[x][x] / float(np.sum(
+                    confusion_mat[x, :]))
+                sensitivity = confusion_mat[x][x] / float(np.sum(
+                    confusion_mat[:, y]))
+                f1_score = 2 * precision * sensitivity / (precision + sensitivity)
+
+                ax.annotate("\n{:4.1f}%\n{:4.1f}% (F)".format(
+                    (confusion_mat[x][y] / float(np.sum(confusion_mat))) * 100,
+                    f1_score * 100),
+                    xy=(y, x + 0.1),
+                    horizontalalignment='center',
+                    verticalalignment='center', fontsize=10,
+                    color=textcolor,
+                    fontweight=this_font_weight)
+
+    # Add values for target correctness etc.
+    for x in range(width):
+        y = len(confusion_mat)
+        correctness = confusion_mat[x][x] / float(np.sum(confusion_mat[x, :]))
+        annotate_str = ""
+        annotate_str += "\n{:5.2f}%".format(correctness * 100)
+        ax.annotate(annotate_str,
+                    xy=(y, x),
+                    horizontalalignment='center',
+                    verticalalignment='center', fontsize=12)
+
+    for y in range(height):
+        x = len(confusion_mat)
+        correctness = confusion_mat[y][y] / float(np.sum(confusion_mat[:, y]))
+        annotate_str = ""
+        annotate_str += "\n{:5.2f}%".format(correctness * 100)
+        ax.annotate(annotate_str,
+                    xy=(y, x),
+                    horizontalalignment='center',
+                    verticalalignment='center', fontsize=12)
+
+    overall_correctness = np.sum(np.diag(confusion_mat)) / np.sum(confusion_mat).astype(float)
+    ax.annotate("{:5.2f}%".format(overall_correctness * 100),
+                xy=(len(confusion_mat), len(confusion_mat)),
+                horizontalalignment='center',
+                verticalalignment='center', fontsize=12,
+                fontweight='bold')
+
+    plt.xticks(range(width), class_names, fontsize=class_names_fontsize,
+               rotation=rotate_col_labels)
+    plt.yticks(np.arange(0, height), class_names,
+               va='center',
+               fontsize=class_names_fontsize, rotation=rotate_row_labels)
+    plt.grid(False)
+    plt.ylabel('Predictions', fontsize=15)
+    plt.xlabel('Targets', fontsize=15)
+
+    # n classes is also shape of matrix/size
+    ax.text(-1.2, n_classes + 0.2, "Specificity /\nSensitivity", ha='center', va='center',
+            fontsize=13)
```

Comment:
```suggestion
    ax.text(-1.2, n_classes + 0.2, "Recall", ha='center', va='center',
            fontsize=13)
```

Specificity/Sensitivity made sense for pathologies but for any classes then this value is recall (could also call sensitivity, but recall maybe better as no association with medicine / sensitivity and specificity)

Reply:
See also https://machinelearningmastery.com/precision-recall-and-f-measure-for-imbalanced-classification/ https://en.wikipedia.org/wiki/Precision_and_recall